ttnn/cpp/ttnn/operations/experimental/reshape/reshape.cpp

```diff
@@ -0,0 +1,133 @@
+
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/common/constants.hpp"
+#include "ttnn/run_operation.hpp"
+#include "reshape.hpp"
+#include "tt_metal/common/constants.hpp"
+#include <functional>
+#include <ttnn/operations/numpy/functions.hpp>
+#include "ttnn/operations/experimental/auto_format/auto_format.hpp"
+#include "ttnn/tensor/tensor_utils.hpp"
+#include "ttnn/operations/data_movement/data_transfer/data_transfer.hpp"
+#include "ttnn/operations/data_movement/slice/slice.hpp"
+#include "ttnn/operations/core/core.hpp"
+
+
+#include "ttnn/tensor/tensor.hpp"
+
+#include <cstdint>
+#include <memory>
+
+#include "common/bfloat16.hpp"
+#include "ttnn/tensor/tensor_impl.hpp"
+#include "ttnn/tensor/tensor_impl_wrapper.hpp"
+#include "ttnn/tensor/tensor_utils.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "tt_metal/common/constants.hpp"
+#include "tt_metal/common/math.hpp"
+#include "tt_metal/third_party/tracy/public/tracy/Tracy.hpp"
+#include "tt_metal/graph/graph_tracking.hpp"
+#include "ttnn/distributed/api.hpp"
+#include "ttnn/distributed/types.hpp"
+#include "ttnn/core.hpp"
+
+
+namespace ttnn{
+
+namespace operations::experimental::reshape {
+ttnn::Tensor tensor_reshape(const ttnn::Tensor& input_tensor, const ttnn::Shape& new_shape) {
+    ZoneScoped;
+    GraphTracker::instance().track_function_start("ttnn::experimental::reshape", input_tensor, new_shape);
+    const auto& new_padded_shape = new_shape.padded_shape();
+    const auto tile = input_tensor.get_tensor_spec().tile();
```

Comment:
I think tile should only be requested for tiled tensors below, and the new tensor created should not be provided with tiling information/ should just assume the default